scripts/dc2/update_truth_table.py

```diff
@@ -0,0 +1,138 @@
+    pixel_scale = args.pixel_size
+    num_pix = args.num_pix
+    # Density profiles
+    kwargs_model = {}
+    kwargs_model['lens_model_list'] = ['SIE', 'SHEAR_GAMMA_PSI']
+    kwargs_model['point_source_model_list'] = ['LENSED_POSITION']
+    arcsec_to_deg = 1/3600.0
+
+    sys_ids = lens_df['lens_cat_sys_id'].unique()
+    progress = tqdm(total=len(sys_ids))
+    for i, sys_id in enumerate(sys_ids):
+        # Slice relevant params
+        lens_info = lens_df[lens_df['lens_cat_sys_id']==sys_id].squeeze()
+        ps_info = ps_df[ps_df['lens_cat_sys_id'] == sys_id].iloc[[0]].copy() # df of length 1
+        ps_info_full = ps_df[ps_df['lens_cat_sys_id'] == sys_id].copy() # of length 2 or 4
+        ps_df_index = ps_info_full[ps_info_full['lens_cat_sys_id'] == sys_id].index # length 2 or 4
+        ps_df.drop(ps_df_index, axis=0, inplace=True) # delete for now, will add back
+        src_light_info = src_light_df.loc[src_light_df['lens_cat_sys_id']==sys_id].iloc[0].squeeze() # arbitarily take the first lensed image, since the source properties are the same across the images
+        z_lens = lens_info['redshift']
+        z_src = src_light_info['redshift']
+        x_lens = lens_info['ra_lens'] # absolute position in rad
+        y_lens = lens_info['dec_lens']
+        # FIXME: further adjustment needed here for the SNe, as the SNe are offset from the host center
+        x_src = ps_info['x_%s' % object_type].values[0] # defined relative to lens center
+        y_src = ps_info['y_%s' % object_type].values[0]
+
+        #######################
+        # Solve lens equation #
+        #######################
+        lens_mass_model = LensModel(['SIE', 'SHEAR_GAMMA_PSI',], cosmo=cosmo, z_lens=z_lens, z_source=z_src)
+        lens_eq_solver = LensEquationSolver(lens_mass_model)
+        kwargs_lens_mass = lensing_utils.get_lens_params(lens_info, z_src=z_src, cosmo=cosmo)
+        x_image, y_image = lens_eq_solver.findBrightImage(x_src, y_src,
+                                                          kwargs_lens_mass,
+                                                          min_distance=0.01, # default is 0.01
+                                                          numImages=4,
+                                                          search_window=num_pix*pixel_scale, # default is 5
+                                                          precision_limit=10**(-10) # default
+                                                          )
+        magnification = np.abs(lens_mass_model.magnification(x_image, y_image, kwargs=kwargs_lens_mass))
+        td_cosmo = TDCosmography(z_lens, z_src, kwargs_model, cosmo_fiducial=cosmo)
+        ps_kwargs = [{'ra_image': x_image, 'dec_image': y_image}]
+        time_delays = td_cosmo.time_delays(kwargs_lens_mass, ps_kwargs, kappa_ext=0.0)
+        n_img = len(x_image)
+
+        #################################
+        # Update lensed AGN truth table #
```

Comment:
We need to update the truth tables for the host galaxies as well. Unless that happens when you are running the postage stamp code?

Reply:
We also need to update the flux values but I can add that in since they involve getting the dust extinction from the milky way using `sims_photUtils`. The flux values are need to make sure we have all values outlined in the truth catalog document: https://docs.google.com/document/d/1PnFZqdMmHBk6ePEvZBdKVOt64vwQ1cse2SaSJTVx7GM/edit?usp=sharing.

Reply:
Ah I see, this is what we've been discussing. No, this is not happening in the update script. Are magnorms the only columns that need updating?